### ISSUE TITLE
Dealing with jQuery when running the app on server

### ISSUE THREAD

**Issue by felipedrumond** [aspnet/JavaScriptServices#199]
I'm having the issue below when running at server my app that uses jQuery. I'm aware that using jQuery is not recommended but at this point, I need to make use of it for a while.

[Node] WARNING: your application is taking a long time to render the application
[Node]
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[0]
      An unhandled exception has occurred while executing the request
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at System.Net.Http.HttpClient.<FinishSendAsync>d__58.MoveNext()

and the html result is

<h1>An unhandled exception occurred while processing the request.</h1>
        <div class="titleerror">TaskCanceledException: A task was canceled.</div>
        <p class="location">ThrowForNonSuccess</p>

As a workaround I can check with  `if (typeof (document) == 'undefined')` or even with ngUniversal.isNode to skip the code that uses jQuery, but I was wondering if there is a better, transparent solution for this. I believe that there should be a configuration at the server to ignore jQuery.

**Comment by laskoviymishka** [comment]
There is no possible way to handle jQuery in a proper way, only ducktapes and workarounds. So if you need a nice code with server render support make sure that you not use any jQuery. But if you still there, and understand consequences of using jQuery with prerendering you need to prepare for some hacks.
Basically there is some options for you, most of them used internally inside of server side prerendering frameworks (a2 and react). 
At the end jQuery is just a wrapper around of DOM, so if you could imitate DOM at node process, than most probably jQuery is also can be there (of course with some limitations). For imitate DOM on server best option right now is using _https://github.com/tmpvar/jsdom_
Here is a snippet how to emulate jQuery on server with this one:
```
 var jsdom = require('jsdom').jsdom;
 var document = jsdom('<html></html>', {});
 var window = document.defaultView;
 var $ = require('jquery')(window);
```
To summarize - it possible, but you should have a really good reason to use it :sweat_smile:

**Comment by felipedrumond** [comment]
Hi laskoviymishka. Thank you very much for your comment.
I ended up skipping jQuery on a server, since I cannot replace some jQuery components for native angular 2 components yet. 

For those who are struggling with jQuery running at server, bellow there is a workaround:

`import * as ngUniversal from 'angular2-universal';`

In your functions that create any jQuery, just check if the code is running at node. If yes, just quit:

` createSomethingWithJQuery()
  { 
    if (ngUniversal.isNode)
        return;`

  ...
}

This is not what I was expecting, but solved the issue.
